lib/ansible/plugins/connection/ssh.py

```diff
@@ -641,18 +643,45 @@ def put_file(self, in_path, out_path):
         # accept them for hostnames and IPv4 addresses too.
         host = '[%s]' % self.host
 
-        if C.DEFAULT_SCP_IF_SSH:
-            cmd = self._build_command('scp', in_path, u'{0}:{1}'.format(host, pipes.quote(out_path)))
-            in_data = None
-        else:
-            cmd = self._build_command('sftp', to_bytes(host))
-            in_data = u"put {0} {1}\n".format(pipes.quote(in_path), pipes.quote(out_path))
-
-        in_data = to_bytes(in_data, nonstring='passthru')
-        (returncode, stdout, stderr) = self._run(cmd, in_data)
-
-        if returncode != 0:
-            raise AnsibleError("failed to transfer file to {0}:\n{1}\n{2}".format(to_native(out_path), to_native(stdout), to_native(stderr)))
+        # since this can be a non-bool now, we need to handle it correctly
+        scp_if_ssh = C.DEFAULT_SCP_IF_SSH
```

Comment:
i would move this into the 'else' below when it does not match 'smart', also probably should be .lower()